src/atoms/IndexAtom.jl

```diff
@@ -9,18 +9,14 @@ mutable struct IndexAtom <: AbstractExpr
     rows::Union{AbstractArray,Nothing}
     cols::Union{AbstractArray,Nothing}
     inds::Union{AbstractArray,Nothing}
+end
 
-    function IndexAtom(
-        x::AbstractExpr,
-        rows::AbstractArray,
-        cols::AbstractArray,
-    )
-        return new((x,), (length(rows), length(cols)), rows, cols, nothing)
-    end
+function IndexAtom(x::AbstractExpr, rows::AbstractArray, cols::AbstractArray)
+    return IndexAtom((x,), (length(rows), length(cols)), rows, cols, nothing)
+end
 
-    function IndexAtom(x::AbstractExpr, inds::AbstractArray)
-        return new((x,), (length(inds), 1), nothing, nothing, inds)
-    end
+function IndexAtom(x::AbstractExpr, inds::AbstractArray)
+    return IndexAtom((x,), (length(inds), 1), nothing, nothing, inds)
 end
```

Comment:
I made these outer constructors to not suppress the creation of the default inner constructor, so I could use it in the bugfix